src/promptflow/promptflow/batch/base_executor_proxy.py

```diff
@@ -0,0 +1,71 @@
+from pathlib import Path
+from typing import Any, Mapping, Optional
+
+import httpx
+
+from promptflow._constants import LINE_TIMEOUT_SEC
+from promptflow.executor._result import AggregationResult, LineResult
+from promptflow.storage._run_storage import AbstractRunStorage
+
+
+class AbstractExecutorProxy:
+    @classmethod
+    def create(
+        cls,
+        flow_file: Path,
+        working_dir: Optional[Path] = None,
+        *,
+        connections: Optional[dict] = None,
+        storage: Optional[AbstractRunStorage] = None
+    ) -> "AbstractExecutorProxy":
+        """Create a new executor"""
+        raise NotImplementedError()
+
+    def destroy(self):
+        """Destroy the executor"""
+        pass
+
+    def exec_line(
+        self,
+        inputs: Mapping[str, Any],
+        index: Optional[int] = None,
+        run_id: Optional[str] = None,
+    ) -> LineResult:
+        """Execute a line"""
+        raise NotImplementedError()
+
+    def exec_aggregation(
+        self,
+        batch_inputs: Mapping[str, Any],
+        aggregation_inputs: Mapping[str, Any],
+        run_id: Optional[str] = None,
+    ) -> AggregationResult:
+        pass
+
+
+class APIBasedExecutorProxy(AbstractExecutorProxy):
```

Comment:
We may not implemented in this PR now, let's implement this when we integrate with Csharp